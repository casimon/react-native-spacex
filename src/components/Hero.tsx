import React from 'react';
import {Image, StyleSheet} from 'react-native';
import images from '../images';

const styles = StyleSheet.create({
  image: {
    width: 100,
    height: 100
  }
});

interface Props {
  url?: string;
}

class Hero extends React.PureComponent<Props> {
  render() {
    const {url} = this.props;
    return url ? (
      <Image
        style={styles.image}
        source={{
          uri: url
        }}
        resizeMode="contain"
      />
    ) : (
      <Image
        style={[styles.image, {borderRadius: 6}]}
        source={images.missing}
        resizeMode="contain"
      />
    );
  }
}

export default Hero;
