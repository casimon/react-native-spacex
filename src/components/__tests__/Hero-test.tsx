/**
 * @format
 */

import 'react-native';
import React from 'react';
import Hero from '../Hero';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders hero correctly with missing image', () => {
  const tree = renderer.create(<Hero />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders hero correctly with patch image', () => {
  const tree = renderer
    .create(<Hero url="https://images2.imgbox.com/12/47/6uim8L1a_o.png" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
