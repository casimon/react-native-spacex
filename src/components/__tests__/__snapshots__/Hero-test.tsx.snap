// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders hero correctly with missing image 1`] = `
<Image
  resizeMode="contain"
  source={
    Object {
      "testUri": "../../../src/images/missing.png",
    }
  }
  style={
    Array [
      Object {
        "height": 100,
        "width": 100,
      },
      Object {
        "borderRadius": 6,
      },
    ]
  }
/>
`;

exports[`renders hero correctly with patch image 1`] = `
<Image
  resizeMode="contain"
  source={
    Object {
      "uri": "https://images2.imgbox.com/12/47/6uim8L1a_o.png",
    }
  }
  style={
    Object {
      "height": 100,
      "width": 100,
    }
  }
/>
`;
